File: SDKs/googleapis.com/androiddeviceprovisioning/v1/typescript/src/sdk/models/shared/customerapplyconfigurationrequest.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { DeviceReference } from "./devicereference";



// CustomerApplyConfigurationRequest
/** 
 * Request message for customer to assign a configuration to device.
**/
export class CustomerApplyConfigurationRequest extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=configuration" })
  configuration?: string;

  @SpeakeasyMetadata({ data: "json, name=device" })
  device?: DeviceReference;
}

File: SDKs/googleapis.com/androiddeviceprovisioning/v1/python/src/sdk/models/shared/customerapplyconfigurationrequest.py
from dataclasses import dataclass, field
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class CustomerApplyConfigurationRequest:
    r"""CustomerApplyConfigurationRequest
    Request message for customer to assign a configuration to device.
    """
    
    configuration: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('configuration') }})
    device: Optional[DeviceReference] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('device') }})
    
